File: src/database/supabase/storage/paymentStorage.js
import { supabase, BUCKETS } from '../config';

/**
 * UPLOAD PAYMENT RECEIPT
 */
export const uploadPaymentReceipt = async (orderId, file) => {
  try {
    const timestamp = Date.now();
    const fileName = `${orderId}/receipt_${timestamp}.jpg`;
    
    const { data, error } = await supabase.storage
      .from(BUCKETS.PAYMENT_RECEIPTS)
      .upload(fileName, file, {
        cacheControl: '3600',
        upsert: false
      });

    if (error) throw error;

    const { data: publicUrlData } = supabase.storage
      .from(BUCKETS.PAYMENT_RECEIPTS)
      .getPublicUrl(fileName);

    return { 
      success: true, 
      url: publicUrlData.publicUrl 
    };
  } catch (error) {
    return { 
      success: false, 
      error: error.message 
    };
  }
};

/**
 * DELETE PAYMENT RECEIPT
 */
export const deletePaymentReceipt = async (receiptUrl) => {
  try {
    // Extract file path from URL
    const urlParts = receiptUrl.split('/');
    const fileName = urlParts.slice(-2).join('/');
    
    const { error } = await supabase.storage
      .from(BUCKETS.PAYMENT_RECEIPTS)
      .remove([fileName]);

    if (error) throw error;

    return { success: true };
  } catch (error) {
    return { 
      success: false, 
      error: error.message 
    };
  }
};

File: app/src/main/java/com/example/bunnix/database/supabase/storage/PaymentStorage.kt
package com.example.bunnix.database.supabase.storage

import android.content.Context
import android.net.Uri
import com.example.bunnix.database.config.SupabaseConfig
import io.github.jan.supabase.storage.storage
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import java.io.File

object PaymentStorage {

    private val storage = SupabaseConfig.client.storage

    // UPLOAD PAYMENT RECEIPT
    suspend fun uploadPaymentReceipt(
        context: Context,
        orderId: String,
        imageUri: Uri
    ): Result<String> = withContext(Dispatchers.IO) {
        try {
            val file = uriToFile(context, imageUri)
            val fileBytes = file.readBytes()
            val timestamp = System.currentTimeMillis()
            val path = "$orderId/receipt_$timestamp.jpg"

            val bucket = storage.from(SupabaseConfig.Buckets.PAYMENT_RECEIPTS)

            // ✅ CORRECT: Using options builder
            bucket.upload(
                path = path,
                data = fileBytes,
                upsert = false
            )

            val publicUrl = bucket.publicUrl(path)
            file.delete()

            Result.success(publicUrl)
        } catch (e: Exception) {
            Result.failure(e)
        }
    }

    private fun uriToFile(context: Context, uri: Uri): File {
        val inputStream = context.contentResolver.openInputStream(uri)
            ?: throw Exception("Cannot open URI")

        val tempFile = File(context.cacheDir, "temp_${System.currentTimeMillis()}.jpg")
        tempFile.outputStream().use { output ->
            inputStream.copyTo(output)
        }
        inputStream.close()

        return tempFile
    }
}